pkg/blobstore/cloud_blob_access.go

```diff
@@ -69,10 +72,18 @@ func (ba *cloudBlobAccess) Put(ctx context.Context, digest digest.Digest, b buff
 func (ba *cloudBlobAccess) FindMissing(ctx context.Context, digests digest.Set) (digest.Set, error) {
 	missing := digest.NewSetBuilder()
 	for _, blobDigest := range digests.Items() {
```

Comment:
What would really improve performance is if we could somehow use pipelining for these requests. Right now we have O(n) roundtrips.

Reply:
Easy enough.

Writing a test for it was rather more effort, but also done.